pkg/odo/cli/component/common_push.go

```diff
@@ -44,6 +45,21 @@ func NewCommonPushOptions() *CommonPushOptions {
 	}
 }
 
+//InitConfigFromContext initializes localconfiginfo from the context
+func (cpo *CommonPushOptions) InitConfigFromContext() error {
+	var err error
+	cpo.LocalConfigInfo, err = config.NewLocalConfigInfo(cpo.componentContext)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+//AddContextFlag adds the context flag to specified command storing value of flag in options.componentContext
+func (cpo *CommonPushOptions) AddContextFlag(cmd *cobra.Command) {
+	genericclioptions.AddContextFlag(cmd, &cpo.componentContext)
```

Comment:
Why are you wrapping one single function to another?

Reply:
This is so that I do not have to expose the context flag variable itself outside of common push

Reply:
I was considering something similar for localconfiginfo exposure